fuel-vm/src/interpreter/blockchain.rs

```diff
@@ -498,9 +498,18 @@ where
 
         self.input_contracts.check(&contract_id)?;
 
+        let length_unpadded: usize = length_unpadded
+            .try_into()
+            .map_err(|_| PanicReason::MemoryOverflow)?;
+
         // Fetch the storage contract
-        let contract = super::contract::contract(self.storage, &contract_id)?;
-        let contract = contract.as_ref().as_ref();
+        let contract_bytes: Vec<u8> =
+            super::contract::contract(self.storage, &contract_id)?
+                .into_owned()
+                .into();
+        let contract_len = contract_bytes.len();
```

Comment:
It would be nice to charge the user here before we do the main work.

Reply:
I'll look into it. This would be quite the architectural change, I think, but if it's how it needs to work, then I'll make it work.

Reply:
K. I split the function in half to return the bytes and then copy them separately. I don't love the solution becuase the functions are doing too much, but it's probably better :)